resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshotsStoreImpl.scala

```diff
@@ -94,7 +95,9 @@ private[spark] class ExecutorPodsSnapshotsStoreImpl(
 
   override def stop(): Unit = {
     pollingTasks.asScala.foreach(_.cancel(false))
-    ThreadUtils.shutdown(subscribersExecutor)
+    // SPARK-41136: it may be called by MGR, and since that MGR has a default 30s timeout, here
```

Comment:
- In this case, we don't need JIRA id in the code. 
- Please write full name instead of MGR.
- When MGR has non-default timeout like 20s, what happens?

Reply:
Thanks, will make the hardcoded 20s configurable, and update and move the comment into 
configuration description.